README.md

```diff
@@ -1 +1,245 @@
-# Visibility of Properties
\ No newline at end of file
+# Property Visibility
+
+## Objectives
+
+1. Connect property visibility to real-world privacy.
+2. Learn the purpose of class file duality in Objective-C (`.h` & `.m`), and the nature of the private `@interface` section.
+3. Identify the three implicit components of using the `@property` syntax (instance variable, setter, getter).
+4. Use the `readonly` and `readwrite` property options to protect a property's setter while making its value publicly visible.
+
+## Public vs Private
+
+So far, all of the properties and methods that we've either shown to you in examples or had you write in labs have been *public*. Like playing around in the open of your front yard, this means that everything you've been doing so far has been *visible* to outside classes. Much of the time, this can be appropriate—but not always.
+
+![](https://upload.wikimedia.org/wikipedia/en/1/18/Private_Property_Sign_on_Fence.jpg)  
+—*"Private Property" sign on a chain link fence*, David Lottito, 2010. [Wikipedia Commons](https://en.wikipedia.org/wiki/File:Private_Property_Sign_on_Fence.jpg)
+
+Just like closing the curtains in your house because what you do at home simply isn't anyone else's business, sometimes it's just appropriate to keep a property or method *private*—meaning that it's only accessible within the class that it belongs to. The primary usefulness in programming of keeping properties and methods private is to avoid accidental changes to information without going through the proper channels; it is, in a way, a form of assigning *jurisdiction* over information and behaviors. 
+
+**Advanced:** *Privatizing properties and methods has the added effect of improving security within your application because it leaves fewer opportunities to generate exploits—hacks that take advantage of other programs and their data.*
+
+## `@interface`: `.h` vs `.m`
+
+You've likely wondered by this point why it is that Objective-C requires *two* files for every class. What's the purpose behind the `.h` and the `.m` files? It sure feels redundant some of the time.
+
+Well, the answer to that is **visibility**. Everything contained in the `.h` header file is what's publicly visible. The `@interface` section is just that: a declaration of the available "interface" for interacting with that class or an instance of it. The interface is, in effect, the "control panel" for that class. It's the set of commands and input/output that's available to the *public*, meaning outside the class's own inner workings. Those inner workings of how a class actually functions is its *implementation* (how it accomplishes what it claims to in its interface) which it should keep *private*. 
+
+Imagine, as an example, an old analog car radio:
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/c/c6/Car_Radio_of_Analog_Era.jpg/1280px-Car_Radio_of_Analog_Era.jpg)
```

Comment:
ah, i need put that image on S3. I'm only linking to images that have creative commons licenses.